packages/typespec-azure-resource-manager/src/private.decorators.ts

```diff
@@ -350,3 +350,12 @@ function hasProperty(program: Program, model: Model): boolean {
   if (model.baseModel) return hasProperty(program, model.baseModel);
   return false;
 }
+
+export function $azureResourceBase(context: DecoratorContext, resourceType: Model) {
+  context.program.stateMap(ArmStateKeys.azureResourceBase).set(resourceType, true);
+}
+
+export function isAzureResourceBase(program: Program, resourceType: Model): boolean {
```

Comment:
isn't it `isAzureResource` more accuratE?>

Reply:
+1 - not sure what the distinction would be with `base`